www/src/public/js/controllers/builder/main.js

```diff
@@ -1196,6 +1328,25 @@
 
                 return;
             }
+
+            //if the new item is not a runecharm, set that charm slot to default AAAAA
+            if($scope.selectedList.items[$scope.currentItemIndex].id === $scope.runeCharmId) {
```

Comment:
You should never be able to click on an item from the item search modal that has that ID. If that does ever happen and that would mean somehow an item was added to the DB with an ID of -5, which would break everything :D. I can add this check though, what do you think?

Reply:
Before, you had: `if($scope.selectedList.items[$scope.currentItemIndex].id === $scope.runeCharmId && angular.copy(item).id != $scope.runeCharmId) {`, which is checking to see if the current item is a rune charm, but the new item is not a rune charm. I think you only need to get rid of the `angular.copy` call, but I think you still need the second expression in the if statement.

So, I think the new line of code should read: `if($scope.selectedList.items[$scope.currentItemIndex].id === $scope.runeCharmId && item.id != $scope.runeCharmId) {`, right? You got rid of the `item.id != $scope.runeCharmId` expression, so it will do this any time the current item is a rune charm.

Reply:
I added the `&&item.id != $scope.runeCharmId` check.